app/models/study.rb

```diff
@@ -1192,6 +1192,28 @@ def default_color_profile
     self.default_options[:color_profile].presence
   end
 
+  # array of names of annotations to ignore the unique values limit for visualizing
+  def override_viz_limit_annotations
+    self.default_options[:override_viz_limit_annotations] || []
+  end
+
+  # make an annotation visualizable despite exceeding the default values limit
+  def add_override_viz_limit_annotation(annotation_name)
+    cell_metadatum = self.cell_metadata.find_by(name: annotation_name)
+    if cell_metadatum
+      # we need to populate the 'values' array, since that will not have been done at ingest
+      uniq_vals = cell_metadatum.concatenate_data_arrays(annotation_name, 'annotations').uniq
+      cell_metadatum.update!(values: uniq_vals)
```

Comment:
Blocker: the reason we stopped doing this is because some annotations (like `biosample_id`) can have 100K+ values, and can eventually blow out the MongoDB 16MB document size limit.  We saw this with Eran's data, which was when we introduced that change.  Unfortunately, we're going to simply have to read these out of the database every time and not store them in the `values` array.  Given the potentially low amount of times this will be used, I think that tradeoff is acceptable.

Reply:
that's a valid concern.  The problem with reading them every time is that for Jonah's study, that's reading a 10MM value array into memory.  Which, based on what we've seen from Jean's testing, is a ~45-second operation.  And that's during the initial fetch of explore data, so that's a 45-second wait before a user gets to even start waiting for their first graph to load.   So, I think our options are (1) assuming that any column that we run this on will fit within the 16MB limit (we'll only be running it on columns specifically requested by the study author, so we don't have to worry about, e.g. biosample_id) or (2) finding some alternate place to store these arrays -- maybe just park it in a file somewhere?    I'd lean towards (1), at least until we confirm it won't work for Jonah's study, and then switch to (2) if it doesn't